refresh.template.py

```diff
@@ -212,36 +215,105 @@ def _get_cpp_command_for_files(compile_action: json):
     command = ' '.join(args) # Reformat options as command string
     return source_files, header_files, command
 
-
-if __name__ == '__main__':
-    # Load aquery's output from the proto data being piped to stdin
-    # Proto reference: https://github.com/bazelbuild/bazel/blob/master/src/main/protobuf/analysis_v2.proto
-    aquery_output = json.loads(sys.stdin.buffer.read(), object_hook=lambda d: SimpleNamespace(**d)) # object_hook allows object.member syntax, just like a proto, while avoiding the protobuf dependency
+def extract(directory: pathlib.Path, aquery_output):
+    """
+    Input (stdin): jsonproto output from aquery, pre-filtered to (Objective-)C(++)
+        compile actions for a given build.
+    Output (stdout): Corresponding entries for a compile_commands.json, with commas after each entry, describing all ways every file is being compiled.
+        Also includes one entry per header, describing one way it is compiled (to work around https://github.com/clangd/clangd/issues/123).
+
+    Crucially, this de-Bazels the compile commands it takes as input, leaving
+    something clangd can understand. The result is a command that could be run from
+    the workspace root directly, with no bazel-specific compiler wrappers,
+    environment variables, etc.
+    """
```

Comment:
I know you may disagree, but could we not hard-wrap this?
[Normally I'd just reformat rather than bother you.]

Reply:
Done.